wp-content/plugins/tapestry/controller/class.user-controller.php

```diff
@@ -0,0 +1,141 @@
+<?php
+// TODO Change exceptions to using an ERROR class
+/**
+ * Add/update/retrieve User progress
+ * 
+ */
+class TapestryUserController {
+
```

Comment:
I think it would be better if you save the current user id to a class property when the class is constructed so you can use it anywhere in your class. You may want to declare functions as static if they don't need the class property.

Reply:
Added User Id as class property.